### PR TITLE
Implemented LookupType Controller. Added 'update' method to LookupType DAO

### PR DESCRIPTION
**Pull request by rma-bryson**

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            LookupTypeDao dao = new LookupTypeDao(dsl);
+            List<LookupType> lookupTypes = dao.retrieveLookupTypes(category, prefix, officeId);
+            LookupType lookupType = lookupTypes.stream()
+                    .filter(lt -> lt.getDisplayValue().equals(displayValue))
+                    .findFirst()
+                    .orElse(null);
+            if (lookupType != null) {
+                dao.deleteLookupType(category, prefix, lookupType);
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java]: Does this call not fail in a manageable way and that's why you're doing the lookup first?

Comment (rma-bryson) [reply cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java]: Refactored the retrieval into the DAO's delete method

**Review by MikeNeilson — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+        requestResultSize = this.metrics.histogram((name(className, RESULTS, SIZE)));
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    @OpenApi(
+            queryParams = {
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java:73]: Need to add the "required = true" for all the required parameters. Check all the methods

File: cwms-data-api/src/test/java/cwms/cda/api/LookupTypeControllerIT.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+        // Structure of test:
+        // 1)Create the LookupType
+        // 2)Retrieve the LookupType and assert that it exists
+        // 3)Delete the LookupType
+        // 4)Retrieve the LookupType and assert that it does not exist
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+
+        //Create the lookup type
+        given()
```
Comment (adamkorynta) [inline cwms-data-api/src/test/java/cwms/cda/api/LookupTypeControllerIT.java:68]: The formatting for these restassured tests have specific indenting - given, when, then assertThat all in the same column and everything else a single tab indent.

**Review by adamkorynta — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java (hunk 2 above)
Comment (rma-bryson) [reply cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java:73]: added

File: cwms-data-api/src/test/java/cwms/cda/api/LookupTypeControllerIT.java (hunk 3 above)
Comment (rma-bryson) [reply cwms-data-api/src/test/java/cwms/cda/api/LookupTypeControllerIT.java:68]: reformatted

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**